frames/stream_frame.go

```diff
@@ -70,11 +71,7 @@ func ParseStreamFrame(r *bytes.Reader) (*StreamFrame, error) {
 	}
 	if dataLen != 0 {
 		frame.Data = make([]byte, dataLen)
-		n, err := r.Read(frame.Data)
-		if n != int(dataLen) {
-			return nil, errors.New("BUG: StreamFrame could not read dataLen bytes")
-		}
-		if err != nil {
+		if _, err := io.ReadFull(r, frame.Data); err != nil {
```

Comment:
I don't like this particular change as it gets rid of a helpful error message in favor of a generic EOF, but we don't have helpful messages in many places where we should…